crates/topos-p2p/src/behaviour/transmission/codec.rs

```diff
@@ -63,9 +113,22 @@ impl Codec for TransmissionCodec {
     where
         T: AsyncWrite + Unpin + Send,
     {
-        write_length_prefixed(io, req.0).await?;
-        io.close().await?;
+        let req = match req {
+            TransmissionRequest::Transmission(req) => req,
+            TransmissionRequest::Synchronizer(req) => req,
+        };
+
+        {
+            // Write the length first
+            let mut buffer = unsigned_varint::encode::usize_buffer();
+            io.write_all(unsigned_varint::encode::usize(req.len(), &mut buffer))
+                .await?;
+        }
+
+        // Write the payload.
+        io.write_all(&req).await?;
```

Comment:
If the payload of `TransmissionRequest` is guaranteed to be encoded by `unsigned_varint`, wouldn't the length be the first item anyway? If that is the case, why do we need to explicitly pluck out the length and then rewrite it in two separate operations? Wouldn't it be faster to `write_all` right away?

Reply:
On the transmission protocol, the legacy encoder is `bincode` and the new one is `prost`